src/ProjectSystem/Impl/IO/MsBuildFileSystemWatcher.cs

```diff
@@ -215,6 +215,9 @@ public sealed partial class MsBuildFileSystemWatcher : IDisposable {
 
             relativePath = PathHelper.MakeRelative(rootDirectory, fullPath);
             try {
+                if (!fileSystem.FileExists(fullPath)) {
```

Comment:
What throws is 

```
                return filter.IsFileAllowed(relativePath, fileSystem.GetFileAttributes(fullPath));

```

specifically

```
fileSystem.GetFileAttributes
```

since the file does not exist.